tests/test_select1.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2017-2018 Snowflake Computing, Inc. All rights reserved.
+ */
+
+#include "utils/test_setup.h"
+#include "utils/TestSetup.hpp"
+
+void test_select1_cpp(void **unused) {
+    Snowflake::Client::Connection *cnx = TestSetup::connectionFactory();
+    SF_STATUS status;
+
+    try {
+        cnx->connect();
+        Snowflake::Client::Statement *stmt = new Snowflake::Client::Statement(*cnx);
+        std::string command("select 1;");
+        stmt->query(command);
+
+        int64 out = 0;
+        SF_BIND_OUTPUT c1 = {
+          idx : 1,
+          c_type : SF_C_TYPE_INT64,
+          max_length : 0,
+          value : (void *) &out,
+          len : sizeof(out),
+          is_null : 0
+        };
+        stmt->bindResult(c1);
```

Comment:
We may want to have a better way for binding. The idea is no C type or even the buffer size is required, but can defer by the time of fetch.

Anyway let's postpone the design to the later stage.

Reply:
I don't know others but think about JDBC interface. It is not necessarily that JDBC is the best, though. More or less object oriented based interface is preferred to leverage C++ feature.